src/what_is_ty_generics.md

```diff
@@ -0,0 +1,18 @@
+# What is `ty::Generics`
+
+The generic parameters introduced by an item are tracked the [`ty::Generics`] struct. Sometimes items allows usage of generics of parent items inside of them, this is accomplished via the `ty::Generics` struct having an optional field to specify a parent item to inherit generic parameters of. For example given the following code:
+
+```rust,ignore
+trait Trait<T> {
+    fn foo<U>(&self);
+}
+```
+
+The `ty::Generics` used for `foo` would contain `[U]` and a parent of `Some(Trait)`. `Trait` would have a `ty::Generics` containing `[Self, T]` with a parent of `None`.
+
+The [`GenericParamDef`] struct is used to represent each individual generic parameter in a `ty::Generics` listing. The `GenericParamDef` struct contains information about the generic parameter, for example its name, defid, what kind of parameter it is (i.e. type, const, lifetime). It also contains a `u32` index representing what position the parameter is (starting from the outtermost parent).
```

Comment:
what's this u32 actually useful for in practice? Also `outtermost` is with one t.